frontend-wasm/src/code_translator/mod.rs

```diff
@@ -0,0 +1,1035 @@
+            let name = mod_info.global_name(global_index);
+            let ty = (&mod_info.globals[global_index]).ty.clone();
+            let ptr = builder
+                .ins()
+                .symbol_addr(name, Ptr(ty.clone().into()), span);
+            let val = state.pop1();
+            builder.ins().store(ptr, val, span);
+        }
+        /********************************* Stack misc **************************************/
+        Operator::Drop => _ = state.pop1(),
+        Operator::Select => {
+            let (arg1, arg2, cond) = state.pop3();
+            // if cond is not 0, return arg1, else return arg2
+            // https://www.w3.org/TR/wasm-core-1/#-hrefsyntax-instr-parametricmathsfselect%E2%91%A0
+            let zero = builder.ins().i32(0, span);
+            let cond_i1 = builder.ins().neq(cond, zero, span);
+            state.push1(builder.ins().select(cond_i1, arg1, arg2, span));
+        }
+        Operator::Unreachable => {
+            builder.ins().unreachable(span);
+            state.reachable = false;
+        }
+        Operator::Nop => {}
+        /***************************** Control flow blocks *********************************/
+        Operator::Block { blockty } => translate_block(blockty, builder, state, mod_info, span)?,
+        Operator::Loop { blockty } => translate_loop(blockty, builder, state, mod_info, span)?,
+        Operator::If { blockty } => translate_if(blockty, state, builder, mod_info, span)?,
+        Operator::Else => translate_else(state, builder, span)?,
+        Operator::End => translate_end(state, builder, span),
+
+        /**************************** Branch instructions *********************************/
+        Operator::Br { relative_depth } => translate_br(state, relative_depth, builder, span),
+        Operator::BrIf { relative_depth } => translate_br_if(*relative_depth, builder, state, span),
+        Operator::BrTable { targets } => translate_br_table(targets, state, builder, span)?,
+        Operator::Return => translate_return(state, builder, diagnostics, span)?,
+        /************************************ Calls ****************************************/
+        Operator::Call { function_index } => {
+            translate_call(state, builder, function_index, mod_info, span, diagnostics)?;
+        }
+        /******************************* Memory management *********************************/
+        Operator::MemoryGrow { .. } => {
+            let arg = state.pop1();
+            state.push1(builder.ins().mem_grow(arg, span));
+        }
+        Operator::MemorySize { .. } => {
+            // Return total Miden memory size
+            state.push1(builder.ins().i32(mem_total_pages(), span));
+        }
+        /******************************* Load instructions ***********************************/
+        Operator::I32Load8U { memarg } => {
+            translate_load_zext(I8, I32, memarg, state, builder, span)
```

Comment:
*Bug*: You'll need to use the unsigned type, e.g. `U8`, for any zero-extensions like this. The codegen backend will not allow zero-extending a signed type as it will lose the signedness by doing so. You must use casts otherwise.